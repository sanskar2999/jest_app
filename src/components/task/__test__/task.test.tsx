import React from 'react';
import { render, screen ,cleanup} from '@testing-library/react';
import Task from '../task';
import renderer from 'react-test-renderer';
afterEach(cleanup);

test('should render task',()=>{
    const tasks= { id:1,title:'wash dishes',completed:false};
  render(<Task task={tasks}/>);
  const element=screen.getByTestId('task-1');
  expect(element).toBeInTheDocument();
  expect(element).toHaveTextContent('wash dishes');
});

test('should render task',()=>{
    const tasks= { id:2,title:'make dinner',completed:true};
  render(<Task task={tasks}/>);
  const element=screen.getByTestId('task-2');
  expect(element).toBeInTheDocument();
  expect(element).toHaveTextContent('make dinner');
  expect(element).not.toContainHTML('<h6>');
});

test ('match snapshot',()=>{
    const tasks= { id:1,title:'wash dishes',completed:false};
    const tree =renderer.create(<Task task={tasks}/>).toJSON();
    console.log(tree);
    expect(tree).toMatchSnapshot();

})