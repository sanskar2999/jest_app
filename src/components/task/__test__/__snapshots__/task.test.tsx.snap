// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  data-testid="task-1"
>
   
  <h1>
    wash dishes
  </h1>
</div>
`;
